### PR TITLE
Cookiecutter

### PR DESCRIPTION
**Pull request by JvanderSaag**
This PR introduces the following:

- Repo is structured according to [cookiecutter guidelines](https://drivendata.github.io/cookiecutter-data-science/)
- Pylint and DSLinter is configured, and return perfect score
- Repo still contains main.py for manual execution of the ML pipeline, this will be removed in later editions

### CODE REVIEW

File: docs/AUTHORS.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+=======
+Credits
+=======
+
+Contributors
+------------
+
+* Jelmer van der Saag <jelmervandersaag@me.com>
```
Comment (alemoraru) [inline docs/AUTHORS.rst]: Change authors to use team, rather than individual.

File: docs/CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+A reminder for the maintainers on how to deploy.
+Make sure all your changes are committed (including an entry in HISTORY.rst).
+Then run::
+
+$ bump2version patch # possible: major / minor / patch
+$ git push
+$ git push --tags
+
+Travis will then deploy to PyPI if tests pass.
```
Comment (alemoraru) [inline docs/CONTRIBUTING.rst]: uhm, we are not using TravisCI - how this fit here? Or am I missing something?

File: docs/LICENSE [hunk 3]
```diff
@@ -0,0 +1,22 @@
+MIT License
+
+Copyright (c) 2023, Jelmer van der Saag
```
Comment (alemoraru) [inline docs/LICENSE]: Do not put personal name, but rather team name.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+from setuptools import setup, find_packages
+
+requirements = [ ]
+
+test_requirements = [ ]
+
+setup(
+    author="Jelmer van der Saag",
+    author_email='jelmervandersaag@me.com',
```
Comment (alemoraru) [inline setup.py]: `author` and `author_email` should again not point to an individual.

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        'Development Status :: 2 - Pre-Alpha',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: MIT License',
+        'Natural Language :: English',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+    ],
+    description="The model-training repository of Team 8 for the Release Engineering for Machine Learning (CS4295) course at the TU Delft.",
```
Comment (alemoraru) [inline setup.py]: `Team 8` -> `Team 08`

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    install_requires=requirements,
+    license="MIT license",
+    include_package_data=True,
+    keywords='model_training',
+    name='model_training',
+    packages=find_packages(include=['model_training', 'model_training.*']),
+    test_suite='tests',
+    tests_require=test_requirements,
+    url='https://github.com/JvanderSaag/model_training',
```
Comment (alemoraru) [inline setup.py]: This URL does not exist -> point to organization repo instead.

File: src/__init__.py [hunk 7]
```diff
@@ -0,0 +1,5 @@
+"""Top-level package for model-training."""
+
+__author__ = """Jelmer van der Saag"""
+__email__ = 'jelmervandersaag@me.com'
+__version__ = '0.2.0'
```
Comment (alemoraru) [inline src/__init__.py]: Again, a lot of metadata that does not fit as mentioned above. Also why `__version__ = '0.2.0'` ? We are sitting on a different version. I would align with the version that is currently in the tags, i.e. the one that is automatically generated with new commits. As such, this version should also get automatically incremented with new releases.

**Review by alemoraru — CHANGES_REQUESTED** [review]
Some minor things to fix. Additionally, versioning should also be automatically incremented in a new commit when pushes to main are done. Let's have a chat about this "offline" - also to see how to automatically add linting and others on the pipeline.

**Comment by JvanderSaag** [comment]
Unnecessary configuration files and artifacts from the cookiecutter repo generation process have been removed. Any other mention of individuals have been changed to that of the team

File: src/__init__.py (hunk 7 above)
Comment (JvanderSaag) [reply src/__init__.py]: This got introduced with automatic generation of the cookiecutter repo. It is not necessary for the sucessful execution of the pipeline, so I have removed the email and version metadata as I could not find an easy way to easily align this versioning with our tags.The author metadata is changed to our team name and number

File: setup.py (hunk 6 above)
Comment (JvanderSaag) [reply setup.py]: URL has been changed to organisation repo

File: setup.py (hunk 5 above)
Comment (JvanderSaag) [reply setup.py]: Has been corrected

File: setup.py (hunk 4 above)
Comment (JvanderSaag) [reply setup.py]: Author has been updated to team number, email has been removed

File: docs/LICENSE (hunk 3 above)
Comment (JvanderSaag) [reply docs/LICENSE]: Has been corrected

File: docs/CONTRIBUTING.rst (hunk 2 above)
Comment (JvanderSaag) [reply docs/CONTRIBUTING.rst]: This was automatically generated by cookiecutter and is indeed not useful for us. It has been removed.

File: docs/AUTHORS.rst (hunk 1 above)
Comment (JvanderSaag) [reply docs/AUTHORS.rst]: Authors and contributor sections have been removed. This information is only really required if publishing to a package repository like PyPi, which we will not be doing. It was automatically generated by cookiecutter, and has now been removed.

**Review by alemoraru — APPROVED** [review]

**Pull request merged**